src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -25,11 +33,17 @@ class FilterOptionsConfig {
      */
     this.control = config.control;
 
+    /**
+     * The type of filtering to apply to the options.
+     * @type {string}
+     */
+    this.optionType = config.optionType || OptionTypes.STATIC_FILTER;
+
     /**
      * The list of filter options to display with checked status
      * @type {object[]}
      */
-    this.options = config.options;
+    this.options = config.options.map(o => ({ ...o }));
```

Comment:
Minor nit, you can use slice() to make a copy

Reply:
for the record I did NOT know about this tip, but after some testing it looks like slice doesn't make a deep copy of the objects. What I did was
```
let a = [{ a: 123}, {a: 345, b: true}]
let b = a.slice()
b[1].a = 123
console.log(a[1].a === b[1].a)
```

and in this case array `a` was also modified